src/Playwright/Core/Page.cs

```diff
@@ -982,9 +992,9 @@ internal void OnFrameNavigated(Frame frame)
 
         internal void FireResponse(IResponse response) => Response?.Invoke(this, response);
 
-        private Task RouteAsync(RouteSetting setting)
+        private Task RouteAsync(RouteHandler setting)
         {
-            setting.Url = Context.CombineUrlWithBase(setting.Url);
+            setting.Url = Helpers.UrlHelpers.ConstructURLBasedOnBaseURL(Context.Options.BaseURL, setting.Url);
```

Comment:
I am not sure if this line is correct.

```suggestion
```

1. `BrowserContext` does not do it:
https://github.com/microsoft/playwright-dotnet/blob/ac79b4bb5e26be3e4fd33b7a713f2e42ce0e01b6/src/Playwright/Core/BrowserContext.cs#L357-L359
2. It is already called here but with a check for leading `*`:
https://github.com/microsoft/playwright-dotnet/pull/1938/files#diff-9eead3ec1fcbb4945a0bd0376c53196ce1533db4cb6c438509f9cca7847997e3R37
3. It does not work with `Unroute` in some cases where no `*` is used:
https://github.com/microsoft/playwright-dotnet/pull/1935#discussion_r785074031
https://github.com/microsoft/playwright-dotnet/pull/1935/files#diff-639d841671d8aafa409d45a0d8fab0834d702fb8a2ceaf8625970b8a8950812cR116-R146

Reply:
very good find, this should definitely not be there.